File: sdk/python/tolap-store/tolap_store/static_identity_resolver.py
from __future__ import annotations


class StaticIdentityResolver:
    """Static identity resolver for testing.

    Allows pre-configuring user groups and roles.
    """

    def __init__(
        self,
        groups: dict[str, list[str]] | None = None,
        roles: dict[str, list[str]] | None = None,
    ) -> None:
        self._groups: dict[str, list[str]] = groups or {}
        self._roles: dict[str, list[str]] = roles or {}

    def get_groups(self, user_id: str) -> list[str]:
        return self._groups.get(user_id, [])

    def get_roles(self, user_id: str) -> list[str]:
        return self._roles.get(user_id, [])

    def add_user_groups(self, user_id: str, groups: list[str]) -> None:
        self._groups[user_id] = groups

    def add_user_roles(self, user_id: str, roles: list[str]) -> None:
        self._roles[user_id] = roles

File: sdk/typescript/packages/store/src/static-identity-resolver.ts
/**
 * TOLAP Static Identity Resolver
 *
 * A simple identity resolver backed by static Maps, useful for testing.
 */

import type { IdentityResolver } from "./types.js";

export class StaticIdentityResolver implements IdentityResolver {
  private groups = new Map<string, string[]>();
  private roles = new Map<string, string[]>();

  /** Set groups for a user. */
  setGroups(userId: string, groupIds: string[]): void {
    this.groups.set(userId, groupIds);
  }

  /** Set roles for a user. */
  setRoles(userId: string, roleIds: string[]): void {
    this.roles.set(userId, roleIds);
  }

  async getGroups(userId: string): Promise<string[]> {
    return this.groups.get(userId) ?? [];
  }

  async getRoles(userId: string): Promise<string[]> {
    return this.roles.get(userId) ?? [];
  }
}
